homeassistant/components/nest/config_flow.py

```diff
@@ -152,7 +164,7 @@ def extra_authorize_data(self) -> dict[str, str]:
 
     async def async_oauth_create_entry(self, data: dict[str, Any]) -> FlowResult:
         """Complete OAuth setup and finish pubsub or finish."""
-        assert self.is_sdm_api(), "Step only supported for SDM API"
+        assert self.config_mode != ConfigMode.LEGACY, "Step only supported for SDM API"
```

Comment:
Note: This is set to != LEGACY rather than == SDM to make it simpler to support the second SDM flow in the future.